### DISCUSSION TITLE
How can I execute a scheduled task in Nuxt 3

### DISCUSSION THREAD

**Discussion by moyukingS** [nuxt/nuxt#20843] [Questions] [answered]
I have a scheduled task that fetches some public configurations through an API and stores them in Redis. How can I execute this scheduled task in Nuxt3

**Comment by flapili** [comment]
in my opinion this is an XY problem and nuxt is not a scheduler.
take a look at prefect or temporal.io

**Comment by mickeldebs** [comment, marked as answer]
I had this issue and tackled it using nitro plugins,
I already scheduled a cron job to run every 5 seconds and its working fine, so basically I made a new folder called server in my root directory and in that folder another one called plugins.
Added a new typescript file that looks like this:
`server/plugins/scheduler.ts`
```
import cron from 'node-cron'

export default defineNitroPlugin(() => {
    cron.schedule('*/5 * * * * *', () => {
        console.log('Running a task every 5 seconds')
    })
})
``` 
That way, when server starts this method will get called.
Hope this helps

**Reply by moyukingS** [reply]
thanks guys. This is the solution I need. I was originally trying to set up this task within the server's lifecycle, but it failed.

**Comment by CyberCowboy404** [comment]
Why use additional plugin if it's available under the hood with nitro?

https://nitro.unjs.io/guide/tasks#scheduled-tasks

**Reply by brunobelloni** [reply]
its working in production?

**Comment by HideInMatrix** [comment]
<img width="1804" height="126" alt="image" src="https://github.com/user-attachments/assets/b9f2d9fe-16e3-40b8-9162-e44b1a3d8667" />
add the config in your nuxt.config.ts
<img width="672" height="318" alt="image" src="https://github.com/user-attachments/assets/5fccefd7-483d-4afb-975b-04b977c40366" />
<img width="498" height="200" alt="image" src="https://github.com/user-attachments/assets/9ffef1e3-5b71-42f6-93c7-af654935f4a7" />
<img width="952" height="404" alt="image" src="https://github.com/user-attachments/assets/5d96275a-1aa2-44ed-aba7-6d09094a19cd" />

**Reply by brunobelloni** [reply]
its working on cloudflare workers 👍
